[clang][deps][cas] Fix include-tree contents with -working-directory

benlangmuir opened the pull request:
The first commit is https://github.com/llvm/llvm-project/pull/84525 and will be rebased before merging.

---
When using -working-directory we were inconsistent about using relativeor absolute paths, which was causing failures to find files in the include-tree filesystem. Now we should be consistently using theabsolute path.

Note: this changes behaviour of things like `__FILE__`, which would normally have a relative path even with -working-directory. In the future we may want to find a way to preserve the relative paths, but that has been much harder and the current change at fixes the compilation failures.
    
rdar://116135029

benlangmuir committed:
[clang][deps] Fix test after upstream change NFC

After the -working-directory fix, this test needs to use a relative path
for "ReadFiles". Note: This test already used a relative path upstream,
but was modified downstream.

benlangmuir committed:
[clang][deps][cas] Fix include-tree contents with -working-directory

When using -working-directory we were inconsistent about using relative
or absolute paths, which was causing failures to find files in the
include-tree filesystem. Now we should be consistently using the
absolute path.

Note: this changes behaviour of things like __FILE__, which would
normally have a relative path even with -working-directory. In the
future we may want to find a way to preserve the relative paths, but
that has been much harder and the current change at fixes the
compilation failures.

rdar://116135029

benlangmuir commented:
@swift-ci please test llvm

cachemeifyoucan reviewed (approved):
Only reviewing the last commit in the PR. LGTM

benlangmuir committed:
[clang][deps] Fix dependency scanning with -working-directory (#84525)

Stop overriding -working-directory to CWD during argument parsing, which
should no longer necessary after we set the VFS working directory, and
set FSOpts correctly after parsing arguments so that working-directory
behaves correctly.

(cherry picked from commit 083da46ff07170471f8bb9ed2947f6ebc725670b)

benlangmuir commented:
@swift-ci please test llvm
